aws4.js

```diff
@@ -27,6 +27,11 @@ function RequestSigner(request, credentials) {
   this.request = request
   this.credentials = credentials || this.defaultCredentials()
 
+  // ES's hostParts are the other way round
+  // E.g. search-cluster-name-aaaa00aaaa0aaa0aaaaaaa0aaa.us-east-1.es.amazonaws.com
```

Comment:
Was actually already supported (it only reversed if the value expected to be `region` was `es` — which can only happen in the `http://search-domainname-domainid.us-east-1.es.amazonaws.com` case…  But I've added an extra test :smile: